src/sync.ts

```diff
@@ -239,6 +246,27 @@ export class SyncApi {
                 RoomMemberEvent.Membership,
             ]);
         });
+
+        room.currentState.on(RoomStateEvent.Marker, function(event, state) {
+            const isValidMsc2716Event =  MSC2716_ROOM_VERSIONS.includes(room.getVersion()) ||
+                // MSC2716 is supported in all existing room versions but special
+                // meaning should only be given to "insertion", "batch", and
+                // "marker" events when they come from the room creator
+                event.getSender() === room.getRoomCreator();
+            console.log(`On marker state isValidMsc2716Event=${isValidMsc2716Event}, event_id=${event.getId()} room.getLastMarkerEventIdProcessed()=${room.getLastMarkerEventIdProcessed()}`);
+            if(
+                isValidMsc2716Event &&
+                // We only need to throw the timeline away once when we see a
+                // marker. All of the historical content will be in the
+                // `/messsages` responses from here on out.
+                event.getId() !== room.getLastMarkerEventIdProcessed()
+            ) {
+                // Saw new marker event, refreshing the timeline
+                console.log('Saw new marker event, refreshing the timeline');
+                room.resetLiveTimeline(null, null);
+                room.setLastMarkerEventIdProcessed(event.getId())
+            }
+        });
```

Comment:
TODO: We don't want to refresh the timeline when:

 - [ ] it's the persons first time syncing the room (and got the state for the first time). They won't have any old events cached so there is nothing to refresh
 - [ ] Someone is doing a `syncFromCache` and seeing the state that way. We already processed the marker event state that was in the cache
    - We could solve this by saving `lastMarkerEventIdProcessed` in localStorage so it's persistent across browser refreshes.

Reply:
Addressed in https://github.com/matrix-org/matrix-js-sdk/pull/2282/commits/ffbcbff97fc9f08136d0bdd18e956051e3c93feb